lib/Util.ts

```diff
@@ -107,17 +107,17 @@ export async function decompressGzippedContent(buffer: Buffer, charset?: string)
  * @return {RegExp}
  */
 export function buildProxyBypassRegexFromEnv(bypass : string) : RegExp {
-    try {
-        // We need to keep this around for back-compat purposes
-        return new RegExp(bypass, 'i')    
-    }
-    catch(err) {
-        if (err instanceof SyntaxError && (bypass || "").startsWith("*")) {            
-            let wildcardEscaped = bypass.replace('*', '(.*)');
-            return new RegExp(wildcardEscaped, 'i');
-        }
-        throw err;
+    let searchRegExpToReplaceSpecialChars = new RegExp('([.])', 'g');
+
+    // replace all . symbols in string by \. because point is a special character
+    let safeRegex = (bypass || "").replace(searchRegExpToReplaceSpecialChars, '\\$1');
+    
+    // check if expression starts with asterisk and replace it with .*
+    if (safeRegex && safeRegex.startsWith("*")) {
+        safeRegex = safeRegex.replace("*", ".*");
     }
+
+    return new RegExp(safeRegex, 'i');
```

Comment:
Could it make sense to expand our test coverage for this function?

https://github.com/microsoft/typed-rest-client/blob/9397d2e164afe5f70119dde75935dba9d09d7afd/test/units/utiltests.ts#L17-L44

Reply:
Yeah, make sense

Reply:
Added several tests that cover cases that I am trying to fix by this PR